### PR TITLE
RUBY-1926 FLE Spec Tests

### PR DESCRIPTION
**Pull request by egiurleo**

### CODE REVIEW

File: lib/mongo/protocol/msg.rb [hunk 1]
```diff
@@ -201,8 +201,8 @@ def maybe_encrypt(server, client)
         if client && client.encrypter && client.encrypter.encrypt?
           if server.max_wire_version < 8
             raise Error::CryptError.new(
-              "Cannot perform encryption against a MongoDB server older than " +
-              "4.2 (wire version less than 8). Currently connected to server " +
+              "Auto-encryption requires a minimum MongoDB version of 4.2 " +
```
Comment (egiurleo) [inline lib/mongo/protocol/msg.rb]: Spec requires a specific message.

File: spec/runners/crud/requirement.rb [hunk 2]
```diff
@@ -42,7 +42,7 @@ def satisfied?
           ok &&= cc.fcv_ish >= short_min_server_version
         end
         if max_server_version
-          ok &&= cc.server_version <= max_server_version
+          ok &&= cc.short_server_version <= max_server_version
```
Comment (egiurleo) [inline spec/runners/crud/requirement.rb]: I changed this to be consistent with how the CRUD spec runner decides whether to run or skip tests: [https://github.com/mongodb/mongo-ruby-driver/blob/master/spec/runners/crud.rb#L209](https://github.com/mongodb/mongo-ruby-driver/blob/master/spec/runners/crud.rb#L209)

File: spec/spec_tests/data/client_side_encryption/updateOne.yml [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+          aws: {} # Credentials filled in from environment.
+    operations:
+      - name: updateOne
+        arguments:
+          filter: { }
+          update: { $rename: { encrypted_string: "random" } }
+        result:
+          errorContains: "$rename between two encrypted fields must have the same metadata or both be unencrypted"
+  - description: "an invalid update (no $ operators) is validated and errors"
+    skipReason: "The Ruby Driver supports this kind of update command"
```
Comment (egiurleo) [inline spec/spec_tests/data/client_side_encryption/updateOne.yml]: I believe that this is a valid update on the Ruby Driver. Is that true?

File: spec/support/utils.rb [hunk 4]
```diff
@@ -278,4 +316,44 @@ def int64_value(value)
 
     uri
   end
+
+  # Client-Side encryption tests introduce the $$type syntax for determining
+  # equality in command started events. The $$type key specifies which type of
+  # BSON object is expected in the result. If the $$type key is present, only
+  # check the class of the result.
+  def match_with_type?(expected, actual)
```
Comment (egiurleo) [inline spec/support/utils.rb]: I'm open to feedback/direction on defining this matcher.

**Comment by p-mongo** [comment]
Superseded by https://github.com/mongodb/mongo-ruby-driver/pull/1751

**Pull request closed without merging**